### PR TITLE
Added HVS MongoDB Atlas integration resource

### PR DESCRIPTION
**Pull request by maxcoulombe**
### :hammer_and_wrench: Description

+ Added a new resource for the HVS MongoDBAtlas Integration
* Pattern largely based on #1078 

### :building_construction: Acceptance tests

- [x] Are there any feature flags that are required to use this functionality?
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:

Screenshot from 2024-09-05 15-32-35

### CODE REVIEW

File: examples/resources/hcp_vault_secrets_integration_mongodbatlas/resource.tf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+resource "hcp_vault_secrets_integration_mongodbatlas" "example" {
+  name         = "my-mongodbatlas-1"
+  capabilities = ["ROTATION"]
+  static_credential_details = {
+    api_secret_key = "12345678-1234-1234-1234-12345678"
```
Comment (averche) [inline examples/resources/hcp_vault_secrets_integration_mongodbatlas/resource.tf:5]:
```suggestion
    api_private_key = "12345678-1234-1234-1234-12345678"
```

File: docs/resources/vault_secrets_integration_mongodbatlas.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+The Vault Secrets MongoDB Atlas integration resource manages an MongoDB Atlas integration.
+
+## Example Usage
+
+```terraform
+resource "hcp_vault_secrets_integration_mongodbatlas" "example" {
+  name         = "my-mongodbatlas-1"
+  capabilities = ["ROTATION"]
+  static_credential_details = {
+    api_secret_key = "12345678-1234-1234-1234-12345678"
```
Comment (averche) [inline docs/resources/vault_secrets_integration_mongodbatlas.md:20]:
```suggestion
    api_private_key = "12345678-1234-1234-1234-12345678"
```

**Review by averche — APPROVED** [review]
🎉 Thanks for putting this together!

File: internal/provider/vaultsecrets/resource_vault_secrets_integration_twilio.go [hunk 3]
```diff
@@ -262,16 +262,9 @@ func (i *IntegrationTwilio) fromModel(ctx context.Context, orgID, projID string,
...
-	}
-	if integrationModel.ResourceName != "" {
-		i.ResourceName = types.StringValue(integrationModel.ResourceName)
-	}
-	if integrationModel.Name != "" {
-		i.Name = types.StringValue(integrationModel.Name)
-	}
+	i.ResourceID = types.StringValue(integrationModel.ResourceID)
+	i.ResourceName = types.StringValue(integrationModel.ResourceName)
+	i.Name = types.StringValue(integrationModel.Name)
```
Comment (maxcoulombe) [inline internal/provider/vaultsecrets/resource_vault_secrets_integration_twilio.go:267]: The bug has been fixed upstream, the update response returns all the fields so we do not need to fallback on the state anymore.

**Review by jasonpilz — APPROVED** [review]
👍🏻

**Pull request merged**